### PR TITLE
plugins/gitlab: init

### LINKED ISSUE #2565: [PLUGIN REQUEST] gitlab.vim
| Field    | Description |
|----------|-------------|
| Plugin   | `gitlab.vim`  | 
| Homepage | `https://docs.gitlab.com/ee/editor_extensions/neovim/`     | 
| Nixpkgs  | `false` | 

## Extra Information
gitlab.vim is an extension which makes it easier to configure gitlab pipelines through neovim. It can be configured with code suggestions through the GitLab language server.

### PR DESCRIPTION
**Pull request by GaetanLepage**
Add support for [gitlab.vim](https://gitlab.com/gitlab-org/editor-extensions/gitlab.vim), GitLab Duo integration in Neovim.

Fixes #2565

### CODE REVIEW

File: plugins/by-name/gitlab/default.nix [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+
+      fix_newlines = defaultNullOpts.mkBool true ''
+        Whether to replace newlines that have become null-byte due to switching between encodings.
+      '';
+
+      lsp_binary_path = defaultNullOpts.mkStr' {
+        pluginDefault = "node";
+        example = lib.literalExpression "lib.getExe pkgs.nodejs";
+        description = ''
+          The path where the language server binary is available or should be installed to.
```
Comment (MattSturgeon) [inline plugins/by-name/gitlab/default.nix:94]: Q: the option relates to node, but the description is talking about a lsp server?

Comment (MattSturgeon) [reply plugins/by-name/gitlab/default.nix:94]: Seems it relates to the runtime that will run the LSP (nodejs), not the LSP itself (gitlab-lsp)

Comment (GaetanLepage) [reply plugins/by-name/gitlab/default.nix]: Rephrased !

**Review by MattSturgeon — APPROVED** [review]
LGTM :rocket:

**Pull request merged**